### DISCUSSION TITLE
AI Development in the Universe

### DISCUSSION THREAD

**Discussion by lumbar527** [endless-sky/endless-sky#8795] [Lore]
It seems that humans do not have very advanced AI. They can make drones, but that's about it. I don't see why they don't have larger AI systems, for to me, ships are controlled fairly simply. Maybe they can't get the AI complicated enough to cover a whole ship.
**I would like to know about AI lore, if there is any.**
If there isn't already, I would say everyone was too busy fighting and stuff to work on AI. I also had the idea of a storyline where you help some Deep scientists work on AI.
**What do other people think about this?**

**Comment by MasterOfGrey** [comment]
The existence and limitations of the Oracle imply some things about AI in this universe, but nothing is firmly stated and I think there are plans to change up the stuff around that.

**Reply by lumbar527** [reply]
Sorry, I need to polish up on FW. What limitations does the Oracle have?

**Reply by lumbar527** [reply]
The Oracle is also a prediction thing, not a piloting thing, which could work differently.

**Reply by bene-dictator** [reply]
Predition / forecasting is exactly the sort of thing an AI would do.

The Deep has a small transport mission that has AI as a major part of its ending scene, and the Vanguard's description includes "automation technologies".

**Reply by lumbar527** [reply]
I guess common AI would just predict on a smaller scale.

**Comment by lumbar527** [comment]
Actually, why doesn't everything have AI in 3014? We are working on self-driving cars now, so I would assume after a thousand years we would have good driving/piloting AI. **Why don't we?**

**Reply by mOctave** [reply]
**Here's my headcanon for AI:**

AI can pilot a ship perfectly well: that's what the autopilot does.
However, people got worried about AI messing up, and so AI piloting was restricted to only be used for situations where a (relatively untrained) human pilot would be unsafe, such as take off/landing. Then, the other half of the autopilot operations in ES are just autopilot plotting a course and you following it, since there's stupid laws from way back in the 23rd century or so preventing AI from being used for other purposes.

Overall, though, AI likely doesn't exist in ES because:
1. ES is a space opera. It's very well done for one, plot-wise, but it's still very, and intentionally, imperfect in its picture of the future.
2. Lore-wise, AI with those capabilities has probably been banned by the Republic after being badly misused, and there might even be rules enforced by the Quarg or Drak regarding AI thanks to the Korath automata.

**Reply by lumbar527** [reply]
Sounds good to me, though I'll wait for other people's ideas before closing.

**Reply by MasterOfGrey** [reply]
**Personally, I'm partial to the headcannon:**

AI works fine, great even. _Networked_ AI is a basket case of inevitable insanity and uncontrollable wanton destruction.

The Syndicate only managed to make the Oracle work, not by networking it to the cloud or whatever and letting it analyse things, but by instead having a *immense* data gathering and reporting operation. This sort of thing would actually be kinda natural for a corporate entity - they live and die by good data reporting - but the trick would be extreme standardisation of data types and reporting metrics.
That way they could feed the Oracle an enormous amount of data, enough to make the sort of predictions that a hyper-advanced networked AI could, but without the drawback of an insane machine going rogue.

Since machine learning requires big data sets to work through, and doing that without a networked data supply would be a headache, that would then justify where the Alpha's might have been key to its development. In essence, they didn't help the Syndicate *just* create an AI, they helped the Syndicate create an AI that didn't need to be *networked* in order to be trained to a useful level of intelligence and insight.

I also like this because massive, hidden-in-plain sight, intelligence scraping from reporting structures is pretty in keeping with the space opera theme (and has recently been showcased in the Star Wars universe in more detail than previously via the Andor show).

**Reply by lumbar527** [reply]
So you're saying that small-scale AI works well, but most large-scale AI go rampant (go insane).

**Reply by MasterOfGrey** [reply]
In essence, I suppose so

**Comment by Zireael07** [comment]
Can the title be amended to "AI development in universe"? Every time this pops up in my feed, I think it's an umpteenth attempt to use real world AI to develop this game...

**Comment by Zitchas** [comment]
It doesn't really explain the lore, but from a meta perspective technology generally follows a more restricted path than it would probably do in real life. That is to say, technology generally doesn't get developed unless there's a reason for it to be developed; and that reasons for it to be developed don't happen unless there's a *story* reason for it to happen. In real life, societies tend to be very broad scientifically speaking, with people haring off in all directions. This tends to mean that, like a well-balanced game of Civilization (Civ 6 or other modern tightly balanced civ-style games), everyone is always at roughly the same technological level. This is great for strategic balance and well-balanced combat challenges, but boring for telling stories; and terrible for writing good science fiction, wherein having contrasts between civilizations is a particularly valuable tool for exploring concepts and differences.

This also ties into the idea that technological progress is not linear. Despite whatever progress humanity might make in the present day towards widespread and general use AI, there's no guarantee that we continue to do so. Whether it just so happens that humanity runs into a roadblock with it and never develops it further; or if we have some kind of crisis wherein we make the choice to utterly and permanently excise AI technology from our society; the practical result is that AI is not particularly advanced within humanity, and generally seems to be viewed with mistrust. Firefly/Serenity seems to be an example of the former; whereas the Dune series is the latter. Likewise, having another civilization (like the Alphas or Koraths) having advanced AI is no guarantee that humanity would.

From a meta perspective, our current level of AI plus another thousand years of development, would result in an AI that renders human participation in the world utterly irrelevant. Especially for something like piloting a spaceship, where processing high volumes of information, high speed analysis of said data, long range physics modelling, and high speed reactions are essential elements. In such an environment, having a human at the controls is a severe liability with little gains. To quote Joker and EDI from the Mass Effect series, the only benefit a human brings to space combat is that their capability of randomly making sub-optimal decisions that defy the analysis of what would be the best choice. In other words, they can succeed by being bad: making choices that are so bad that the opposing AI doesn't believe that they'd actually do something so bad. Unlike Mass Effect (where ship weapons have generally a low rate of fire but are high speed and high impact), our typical weapons are almost entirely high volume fired weapons with inaccuracy sufficient that dodging and any kind of decision making about a particular shot is irrelevant, as the statistical average of shots generally makes a cone large enough to be almost impossible to dodge a meaningful number of shots unless one manages to dodge *all* the shots. As such, making a sub-optimal choice that leads one to being where the AI expects one to not be is useless. (It is, indeed, ironic that the very measure intended to make our game AI appear to be more human and less precise has the result of devaluing human participation and rendering dodging and personal skill less meaningful.)

Obviously, it's pretty hard to make a game that encourages human participation when the AI is sufficiently good that humans just sit back and let the AI do its thing, so the baseline assumption is that AI is *not* capable of sentient-entity level of performance as a general rule, and that for whatever reason, development thereof is avoided, suppressed, or otherwise not happening for some reason, intentionally or otherwise. This could be something that is explored at greater length; or it could be simply left to be a vague idea that most cultures have either chosen to not develop it, or had some sort of traumatic experience pushing them away from it.

**Reply by lumbar527** [reply]
Maybe there was an age where most things were controlled by AI, but they kept getting too advanced—and weren't stable. They had to be destroyed. Large-scale AI would also very quickly become uncontrollable, and maybe required entire networks to be shut down.

**Reply by Zireael07** [reply]
@lumbar527 Reminds me of the part of Dune universe's backstory

**Reply by lumbar527** [reply]
I haven't read that. My idea of large-scale AI requiring networks to be shut down to destroy them comes from Bungie's Marathon trilogy. It's another old Mac game.

**Reply by Saturnalia9** [reply]
> Maybe there was an age where most things were controlled by AI, but they kept getting too advanced—and weren't stable. They had to be destroyed. Large-scale AI would also very quickly become uncontrollable, and maybe required entire networks to be shut down.

What about remote operations, then? We don't ha e that either, but I see no real reason why it shouldn't exist.

**Comment by Zitchas** [comment]
As potentially relevant, the Android description:

> 	"While these were once considered cutting-edge outfits, their lack of autonomous capabilities and economic benefit ensured they were never licensed by a manufacturer."
	"	Despite this, some captains commissioned these directly from their inventor, effectively replacing one of their crew members with a robotic counterpart. These captains found that androids possessed sufficient capabilities to perform most work, but lacked the capability to captain a ship or take part in boarding actions."

Now, there's a couple caveats here. On the one hand, this "lack of sufficient autonomy" is in regards to replacing an engineer or highly skilled crew member; not just merely driving a car down a street or some other highly defined invariable task.

The other note is that it mentions economic reasons, and that no manufacturer licensed them. They could have been miracle machines as far as we know, but if no manufacturer was willing to license them, then they still wouldn't be used. Now, I'm not saying this is the case, but if a strong luddite movement arose that had large segments of the population intentionally destroying robots and AI whenever found (and by large, I mean something big, like >30% of the population, maybe even the majority.)... then no-one is going to invest in AI and robots will fall into disuse (or at least secretive use).

For instance, look at the reception of AI right now. Some people love it, some people hate it, and a not-insignificant portion of the population is vehemently and aggressively opposed to it as an existential threat to jobs, careers, and even humanity's existence. And that's just to what amounts to a really good chatbot. A not insignificant number of people are fixating on the existential threat of general purpose AI that doesn't even exist yet (and might never) to the point of ignoring all the problems and existential threats that humanity is facing right now. So it isn't unlikely that - if someone in Endless Sky did develop a true general purpose AI at some point in humanity's history, the response was to kill the person(s) responsible, bomb their facility into a crater, and then track down and melt into slag every electronic device and storage medium that any of those people had come within a km of, not to mention every even *potentially* networked device that anyone in said facility might possibly have even thought about networking with. (This is basically the situation in Dune. Albeit in Dune there actually was some historical events to use as justification for such fear and paranoia)

**Reply by Zitchas** [reply]
Also worth noting that slavery still exists; as does indentured servitude and near-slave labor. Robotics and AI are expensive to maintain when one can just drag a body into the incinerator and drop a fresh person into the role.

**Reply by lumbar527** [reply]
I know this might not be good for gameplay, but I simply want a historic explanation. You say: expenses and opinions?

**Reply by Zitchas** [reply]
Personally, I feel the impact on gameplay was a bit overblown and easily mitigated by having automatons require some sort of charging station that takes up a bunk. Likewise, the capture meta is already broken OP, and needs a lot more than just a small limitation in the form of ship erraticity not being avoidable via automatons.

There really is no concrete history to work with here.  That being said, given where humanity is currently at and where the game is; I think the best interpretation is that somehow we arrived at something akin to a Dune situation. Maybe not to the same extreme (If I recall correctly, they didn't even like using *calculators*, let alone computers... But lots of people pushing the edges, like the training combat robot), but some sort of decidedly firmly entrenched anti-AI cultural phenomenon, reinforced by costs. Pretty much has to be a holistic situation, because if it's just cultural, some corporation will just do it anyway. And if it's just expense, some hobbyists or government will do it anyway, perhaps under cover of military tech. The only drones humanity has currently are both Republic Navy only, after all, which indicates that they at least have done a little in this regard.

**Reply by lumbar527** [reply]
Makes sense.

Should I close because this seems to have been answered, or keep open for future discussion?

**Reply by Zitchas** [reply]
I'd say leave it for future discussion. Unlike issues and PRs, discussions don't clog up lists of "things that need to be dealt with".

And really, it's a topic that could stand to have a bit more development. Lots of room for people to think about exactly what happened to have the galaxy in the state that it is in, and what nuggets for story there might be there.

**Comment by Hurleveur** [comment]
https://github.com/endless-sky/endless-sky/pull/8828 shows some navigation AI in the works, idea being that given their limited numbers uhai may use AI to support their crew at some point, but its still not there yet

**Reply by lumbar527** [reply]
The Ladybug that pops up? Good point.

**Reply by Hurleveur** [reply]
I'm thinking of having the AI go rogue but in a benevolent way if you want to know everything...
Also maybe the Locust would be AI assisted to allow it to only have 1 crew-member

**Comment by mOctave** [comment]
ES ships are already ridiculously undercrewed by normal sci-fi standards, which somehow allows for indefinite life support. I wouldn't be surprised if the purpose of crew in most ES ships was to keep everything running while the autopilot did stuff at the (human) pilot's command.

**Reply by Hurleveur** [reply]
Are they tho? The ships are tiny by sci fi standards

**Reply by MasterOfGrey** [reply]
They're *under-massed*. They're not necessarily tiny.

**Reply by Hurleveur** [reply]
By that same reasoning one unit of crew could be 10 crew if one mass is 10 mass
And well looking at the cockpits they seem tiny

It seems that AI is forbidden in republic space?

**Reply by bene-dictator** [reply]
Only AI that resembles a human, I guess – from what I remember in the Riley/Church mission

**Reply by lumbar527** [reply]
Wait, I thought Riley/Church was from one of Midnight's plugins.

**Reply by bene-dictator** [reply]
No, it's in main.

**Reply by lumbar527** [reply]
OK. (The first time I saw it I had his plugins and it's a kind-of weird thing that felt like it would go with the plugins)

Actually, the fact of that being official is very helpful to to discussion—thanks bene-dicator, Hurleveur.
